GitSearch/Application/SearchResult/SearchResultViewController.swift

```diff
@@ -10,65 +10,59 @@ import UIKit
-
-        collectionView?.backgroundColor = #colorLiteral(red: 0.9215686275, green: 0.9215686275, blue: 0.9215686275, alpha: 1)
-        collectionView?.register(SearchResultCell.self, forCellWithReuseIdentifier: "cellId")
+        
+        collectionView?.backgroundColor = .secondaryTint // #colorLiteral(red: 0.9215686275, green: 0.9215686275, blue: 0.9215686275, alpha: 1)
+        collectionView?.register(SearchResultCell.self, forCellWithReuseIdentifier: cellDefault)
         
         let layout = UICollectionViewFlowLayout()
-        layout.estimatedItemSize = CGSize(width: view.frame.size.width-32, height: 120)
+        layout.estimatedItemSize = CGSize(width: view.frame.size.width - 30, height: 120)
         collectionView?.collectionViewLayout = layout
-        
     }
     
     override func numberOfSections(in collectionView: UICollectionView) -> Int {
         return 1
     }
     
     override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return gitRep.items?.count ?? 0
+        return items.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return 0.0
     }
     
     override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cellId", for: indexPath) as! SearchResultCell
-        if let item = getItemAtIndex(index: indexPath.row) {
-            cell.fillInfo(item: item)
-        }
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: cellDefault, for: indexPath) as! SearchResultCell
+        let item = items[indexPath.row]
+        cell.fillInfo(for: item, isLast: items.count == indexPath.row)
+//cell.layer.borderWidth = 1
+//cell.layer.borderColor = UIColor.black.cgColor
         return cell
     }
     
     override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        if let item = getItemAtIndex(index: indexPath.row) {
-            let newVC = RepoInfoViewController(properties: props, item: item)
-            navigationController?.pushViewController(newVC, animated: true)
-        }
-    }
-    
-    func getItemAtIndex(index: Int) -> Dictionary<String, String>! {
-        guard let gitItems = self.gitRep.items else {
-            return nil
-        }
-        return gitItems[index].ToArray(props: &props)
+        let item = items[indexPath.row]
+        let newVC = RepoInfoViewController(item: item)
+        navigationController?.pushViewController(newVC, animated: true)
     }
     
     init(data: SearchInfo) {
+        if let dataItems = data.items {
+            items = dataItems
+        }
```

Comment:
Possible bug here: if `data.items == nil`, you will not refresh `items` ivar. But it should be reset to empty: 
```suggestion
    items = data.items ?? []
```

Reply:
https://github.com/mudritskiy/GitSearch/blob/d2a856f93cb0f53059154b94bf80951370b89d6c/GitSearch/Application/SearchResult/SearchResultViewController.swift#L14
There is default declaration at the beginning, so, if it not passed into `if`it is already `[]`
Or you want to say it is ambiguous way?

Reply:
Well, I proposed this change just because its one-liner instead of 3 lines :)